typescript/deploy/src/router/deploy.ts

```diff
@@ -20,11 +21,11 @@ export abstract class AbacusRouterDeployer<
   Chain extends ChainName,
   Contracts extends RouterContracts,
   Factories extends RouterFactories,
-  Config,
-> extends AbacusDeployer<Chain, Config & RouterConfig, Factories, Contracts> {
+  Config extends RouterConfig,
```

Comment:
ooh nice

Reply:
is it nice tho?

It forces developers to have two types, 1) with RouterConfig type that they use to override RouterDeployer and 2) one without RouterConfig so that it can statically type